dracut/30ignition/ignition-umount.service

```diff
@@ -0,0 +1,12 @@
+[Unit]
+Description=Ignition (umount)
+DefaultDependencies=false
+
+# Only unmount after we're done writing to the filesystems
+Requires=ignition-files.service
+After=ignition-files.service
```

Comment:
Having mount and unmount in two separate units seems kind of awkward and brittle, since all units that run in between will need explicit ordering.  How about the approach that [sysroot-boot.service](https://github.com/coreos/bootengine/blob/107ce54747f739dad4b56f9753852341bb759a4d/dracut/30ignition/sysroot-boot.service) uses, where unmount happens as an `ExecStop` and the service conflicts with things that run during initramfs exit?

Reply:
Does it correctly fail the boot if `ExecStop` fails though? I guess I can try it out and find out. :)

Reply:
That was a good tip!

> Does it correctly fail the boot if `ExecStop` fails though?

I got this to work with:

```
# This ensures that we fail the boot if the umount stage fails
OnFailure=emergency.target
OnFailureJobMode=isolate
```